### PR TITLE
bump create-llama and fix data source and llamacloud service

### PR DESCRIPTION
**Pull request by leehuwuj**

### CODE REVIEW

File: src/ragapp/patch/frontend/app/components/ui/lib/utils.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { clsx, type ClassValue } from "clsx";
+import { twMerge } from "tailwind-merge";
+
+export function cn(...inputs: ClassValue[]) {
+  return twMerge(clsx(inputs));
+}
+
+export function isValidUrl(url?: string): boolean {
```
Comment (leehuwuj) [inline src/ragapp/patch/frontend/app/components/ui/lib/utils.ts:8]: This is good to add in CL, should change to hasUrl() instead, and just use it directly here: https://github.com/run-llama/create-llama/blob/f04f60d555c9edf5f86d0d89a534b4c150887732/templates/types/streaming/nextjs/app/components/ui/chat/index.ts#L116

File: src/ragapp/backend/routers/chat/index.py [hunk 2]
```diff
@@ -84,3 +68,17 @@ async def chat_request(
...
     )
+
+
+def process_response_nodes(
+    nodes: List[NodeWithScore],
+    background_tasks: BackgroundTasks,
+):
+    try:
+        # Start background tasks to download documents from LlamaCloud if needed
+        from app.engine.service import LLamaCloudFileService
```
Comment (leehuwuj) [inline src/ragapp/backend/routers/chat/index.py:79]: As we always call `from app.engine.service import` so the LLamaCloudFileService file must be there, so patching is the only way to go.

File: src/ragapp/patch/frontend/app/components/ui/lib/utils.ts (hunk 1 above)
Comment (marcusschiesser) [reply src/ragapp/patch/frontend/app/components/ui/lib/utils.ts:8]: yes, please try https://github.com/run-llama/create-llama/pull/259 - if it works, i'll release it - then you don't need to patch

File: src/ragapp/patch/backend/app/engine/service.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
+import logging
```
Comment (marcusschiesser) [inline src/ragapp/patch/backend/app/engine/service.py:1]: i see two options:

1. move this to ragapp/backend and reimport it here in the patch (then we would own the file service)
2. reference the exact commit id used from CL like:
```suggestion
# same as https://github.com/run-llama/create-llama/blob/{commit_id}/templates/components/vectordbs/python/llamacloud/service.py
import logging
```

File: src/ragapp/patch/frontend/app/components/ui/lib/utils.ts (hunk 1 above)
Comment (leehuwuj) [reply src/ragapp/patch/frontend/app/components/ui/lib/utils.ts:8]: Please ship it! Thank you!

File: src/ragapp/patch/backend/app/engine/service.py (hunk 3 above)
Comment (leehuwuj) [reply src/ragapp/patch/backend/app/engine/service.py:1]: Thanks for the ideas; the `2.` option looks simpler to me so I'll go with it 😄

**Review by marcusschiesser — APPROVED** [review]

**Pull request merged**